### ISSUE TITLE
Use of try_join_all hurts Tokio performance significantly

### ISSUE THREAD

**Issue by Darksonn** [jebrosen/async-file-benchmark#6]
Due to [the coop system](https://tokio.rs/blog/2020-04-preemption), this is very slow on Tokio due to [this issue](https://github.com/tokio-rs/tokio/issues/2401). The issue was fixed for `join_all`, but not `try_join_all`.

**Comment by jebrosen** [comment]
Ah yes - thanks for bringing this up! I have been aware of the general issues with `*_join_all` and `coop` for a while. However, I have not looked at this particular project in some time - possibly before coop was introduced. Am I right in thinking that `spawn` should avoid those problems? In fact, `spawn` would even be closer to the original scenario than `try_join_all`. I believe I switched to `try_join_all` to avoid abstracting over (and measuring the overhead of) the different `spawn` interfaces and implementations.


So I tried updating and fixing this and other issues, but I concluded it would be better to simply update the README and archive the project. I originally wrote this as a reproduction case for a ~40x performance difference that no longer exists, and I haven't maintained it. I sincerely doubt that it produces (or ever produced?) useful or reliable metrics for real applications.

**Comment by Darksonn** [comment]
Yeah, `spawn` should avoid these issues. If you note the issue in the README, that seems fine to me.

It turns out that there aren't many benchmarks comparing Tokio and async-std, so this is one of the benchmarks people use as it's the only they can find.
